File: Data Structures & Algorithms/same-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p==null && q==null)return true;
         if(p==null || q==null)return false;
        TreeNode temp = p;
        TreeNode temp2 = q;
        if(temp.val!=temp2.val){
            return false;
        }
        boolean lefti = isSameTree(p.left,q.left);
        if(lefti==false)return false;
        boolean righti= isSameTree(p.right,q.right);
        if(righti==false)return false;
        return true;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int goodNodes(TreeNode* root) {
       return dfs(root,root->val);
    }

    int dfs(TreeNode* root,int maxi){
        int res=0;
          if(root==nullptr) return res;
        if(root->val>=maxi)res =1;
        maxi = max(maxi,root->val);
        res+=dfs(root->left,maxi);
        res+=dfs(root->right,maxi);
        return res;
    }
};
